### ISSUE TITLE
Add GPFM as a supported tile encoder

### ISSUE THREAD

**Issue by clemsgrs** [clemsgrs/slide2vec#210]
## Summary

Add **GPFM** (Generalizable Pathology Foundation Model; [birkhoffkiki/GPFM](https://github.com/birkhoffkiki/GPFM), weights [`majiabo/GPFM`](https://huggingface.co/majiabo/GPFM)) as a supported **tile encoder**.

GPFM is a strong, openly-licensed (**MIT**) generalist tile encoder and is the cleanest open addition surfaced from the WAIV histoboard survey. It is a `ViT-L/14` DINOv2 backbone (dim 1024), so it slots into the existing dense-capable `TimmTileEncoder` path — the only twist is that the weights ship as a standalone checkpoint rather than a `timm` `hf-hub:` config.

## Model facts (verified against the HF card + GitHub)

- Arch: `timm` `vit_large_patch14_dinov2.lvd142m`, `img_size=224`, `init_values=1e-5`, `num_classes=0` (DINOv2 ViT-L/14)
- Embedding dim: **1024**; patch_size **14** (dense grid 16×16 at 224)
- Weights: **openly downloadable, not gated**, **MIT** — `hf_hub_download(repo_id="majiabo/GPFM", filename="GPFM.pth")`. NOTE: the card explicitly says it is **not** loadable via `timm.create_model("hf-hub:majiabo/GPFM")`; you must build the arch with `pretrained=False` and `load_state_dict` the checkpoint (the MOOZY pattern — see `slide2vec/encoders/models/moozy/loading.py`).
- Transform: Resize(224) + **ImageNet** norm (mean `(0.485,0.456,0.406)`, std `(0.229,0.224,0.225)`) — matches `TimmTileEncoder.get_transform()` for this backbone, so no custom transform needed.
- Precision: upstream uses plain PyTorch, no autocast mentioned → `precision="fp32"` (same reasoning as mSTAR, #208).
- Magnification/spacing: **40× — ~512×512 px tiles resized to 224** → `supported_spacing_um = 0.25`. Rationale: slide2vec extracts `input_size` px at `supported_spacing_um` (see `resolve_preprocessing_requirements`), so we encode the *training magnification* faithfully (40× = 0.25 µm/px), exactly as `uni` encodes its 20× as `0.5`. **Confirm this 0.25 choice — see verification item.**

## Implementation

Create `slide2vec/encoders/models/gpfm.py` reusing `TimmTileEncoder` but loading the GPFM checkpoint instead of timm-pretrained weights:

```python
"""GPFM tile encoder (Generalizable Pathology Foundation Model).

GPFM (Ma et al., 2024; ``birkhoffkiki/GPFM``) is a ViT-L/14 DINOv2 tile encoder.
Weights are openly available (MIT) at ``majiabo/GPFM`` but ship as a standalone
``GPFM.pth`` checkpoint, so we build the timm arch unpretrained and load it.
"""

import torch
from huggingface_hub import hf_hub_download

from slide2vec.encoders.base import TimmTileEncoder
from slide2vec.encoders.registry import register_encoder


@register_encoder(
    "gpfm",
    output_variants={"default": {"encode_dim": 1024}},
    default_output_variant="default",
    input_size=224,
    patch_size=14,
    supported_spacing_um=0.25,  # 40x; ~512px tiles resized to 224 (per card)
    precision="fp32",           # upstream runs plain fp32, no autocast
    source="majiabo/GPFM",
)
class GPFM(TimmTileEncoder):
    def __init__(self, *, output_variant: str | None = None):
        super().__init__(
            "vit_large_patch14_dinov2.lvd142m",
            output_variant=output_variant,
            pretrained=False,
            img_size=224,
            init_values=1e-5,
            dynamic_img_size=True,
        )
        ckpt = hf_hub_download(repo_id="majiabo/GPFM", filename="GPFM.pth")
        state = torch.load(ckpt, map_location="cpu")
        # GPFM.pth may be a raw state_dict or wrapped (e.g. {"model": ...}/"teacher").
        # Unwrap to the bare state_dict and strip any prefix before load_state_dict.
        self._model.load_state_dict(state, strict=True)
```

- Register `gpfm` in `slide2vec/encoders/models/__init__.py` (import block + `__all__`, alphabetical).
- Add to the supported-models docs: `README.md` tile count, and `docs/models.rst` (tile table + the **dense-grid** and **dense-attention** coverage lists, since it's a dense-capable ViT — mirror what #209 did for `mstar`).
- Add `gpfm` to the metadata-contract test data: `tests/test_encoder_registry.py` (`EXPECTED_TILE_ENCODERS`), `tests/test_patch_size_metadata.py` (`(14, 14)`), `tests/test_regression_core.py` (`list_models("tile")` expectation), plus a focused `test_gpfm_metadata_contract`.

## Verification (resolve before/while implementing; surface to maintainer if ambiguous)

1. **Checkpoint structure** — inspect `GPFM.pth`: confirm whether it is a bare `state_dict` or wrapped (`{"model": ...}`, `{"teacher": ...}`, `module.`/`backbone.` prefixes). Adjust unwrapping so `load_state_dict(strict=True)` succeeds. The encoder must load cleanly. (This is a `@pytest.mark.heavy` real-weight check — do not add it to the default suite.)
2. **Spacing** — confirm `supported_spacing_um=0.25` is the intended convention vs. running GPFM at the house-standard `0.5` (20×). Recommended: 0.25 (faithful to 40× training). Flag to maintainer if unsure; do not silently change.
3. **Output dim** — confirm `model(image)` returns a 1024-d pooled CLS embedding (expected for ViT-L/14 `num_classes=0`).

## Acceptance criteria

- [ ] `slide2vec/encoders/models/gpfm.py` registers encoder `gpfm` with the metadata above (level tile, dim 1024, input_size 224, patch_size 14, supported_spacing_um 0.25, precision fp32, source `majiabo/GPFM`).
- [ ] `gpfm` is registered/importable and the checkpoint-load path is implemented (weights download + `load_state_dict`).
- [ ] Metadata-contract tests pass including GPFM (`test_encoder_registry.py`, `test_patch_size_metadata.py`, `test_regression_core.py`), with a focused `test_gpfm_metadata_contract`. Default suite stays network-free.
- [ ] README + `docs/models.rst` list GPFM as a tile encoder (incl. dense coverage lists).
